Add Cisco CBS350-24XT device type definition

dondundonki opened the pull request:
Greetings!

A small PR to add CBS350-24XT.

dondundonki committed:
Add Cisco CBS350-24XT device type definition

joewesch commented on device-types/Cisco/CBS350-24XT.yaml:
```diff
@@ -0,0 +1,92 @@
+  maximum_draw: 125
+  allocated_draw: null
+interfaces:
+- name: '[te]1/0/1'
```
Are these interfaces really named `[te]`? Most other `10gbase-t` interfaces for Cisco in this library are `TenGigabitEthernet`

dondundonki replied:
Changed!

joewesch commented on device-types/Cisco/CBS350-24XT.yaml:
```diff
@@ -0,0 +1,92 @@
+interfaces:
+- name: '[te]1/0/1'
+  type: 10gbase-t
+  mgmt_only: false
```
Can you please remove `mgmt_only: false` from all non-mgmt ports? It just adds noise to the file as that is the default value.

joewesch commented on device-types/Cisco/CBS350-24XT.yaml:
```diff
@@ -0,0 +1,92 @@
+  type: 10gbase-t
+  mgmt_only: false
+- name: '[te]1/0/21'
+  type: 10gbase-t
```
According to the data sheet in the link you provided in the comment, the last 4 interfaces are 10g copper/sfp combo. I _think_ that is supposed to be `10gbase-x-sfpp`?

dondundonki replied:
When I run show int status I see 10G-Combo-C. Upon further research it seems to be a specific type of port on a network switch that can support either a 10G SFP+ copper connection (often a Direct Attach Cable or DAC) or an optical module so I'm not sure if it should be 10gbase-x-sfpp. Do let me know! Thanks!

joewesch replied:
Let's change it to `10gbase-x-sfpp` to represent SFP+ by default, despite the fact that it can work as both.

dondundonki replied:
Updated! Thanks for guiding!

joewesch commented on device-types/Cisco/CBS350-24XT.yaml:
```diff
@@ -0,0 +1,92 @@
+- name: '[te]1/0/24'
+  type: 10gbase-t
+  mgmt_only: false
+- name: oob
```
Same comment about the name as previous. I do see that we have this name also on one other device ([SG550XG-24F-K9](https://github.com/nautobot/devicetype-library/blob/da3b95aec99bcb1656a13a6505409f54c43df930/device-types/Cisco/SG550XG-24F-K9.yaml#L128)), so I'm just double checking that this is correct?

dondundonki replied:
Yup the interface name is oob.

dondundonki committed:
Update CBS350-24XT.yaml

Remove mgmt_only for the last ports according to show int status I see 10G-Combo-C upon further research it seems to be a specific type of port on a network switch that can support either a 10G SFP+ copper connection (often a Direct Attach Cable or DAC) or an optical module so I think it should still be 10g-base-t but I'm new so let me know!

dondundonki committed:
Merge branch 'main' into cisco-cbs350-24xt

dondundonki committed:
Update CBS350-24XT.yaml

Updated the last four ports to be 10gbase-x-sfpp to denote sfp+
